src/overlays/actors/ovl_En_Kame/z_en_kame.c

```diff
@@ -476,123 +587,157 @@ void func_80AD7E0C(EnKame* this, PlayState* play) {
     Actor_PlaySfx(&this->actor, NA_SE_EN_PAMET_DAMAGE);
     this->collider.base.acFlags &= ~AC_ON;
-    this->actionFunc = func_80AD810C;
+    this->actionFunc = EnKame_Damaged;
 }
 
-void func_80AD810C(EnKame* this, PlayState* play) {
+/**
+ * Plays the damaged animation, then goes back to struggling to flip itself upright.
+ */
+void EnKame_Damaged(EnKame* this, PlayState* play) {
     if (SkelAnime_Update(&this->snapperSkelAnime)) {
-        func_80AD7DA4(this);
+        EnKame_SetupStruggle(this);
     }
 }
 
-void func_80AD8148(EnKame* this, PlayState* play) {
+void EnKame_SetupDead(EnKame* this, PlayState* play) {
     Animation_PlayLoop(&this->snapperSkelAnime, &gSnapperDeathAnim);
     Actor_SetColorFilter(&this->actor, COLORFILTER_COLORFLAG_RED, 255, COLORFILTER_BUFFLAG_OPA, 20);
     this->collider.base.acFlags &= ~AC_ON;
     this->collider.base.atFlags &= ~AT_ON;
     this->collider.base.atFlags &= ~(AC_HARD | AC_HIT);
     this->actor.velocity.y = 15.0f;
     this->actor.speed = 1.5f;
+
+    // This allows callers of this function to skip the "finishing blow" screen flash and game freeze as well
+    // as the rotation that happens if the Snapper is killed by a projectile. Only EnKame_Stunned calls this
+    // while passing NULL instead of play; in practice, this means that the only way to skip this code is to
+    // kill the Snapper with an Ice Arrow, since that's the only attack that both stuns it and deals damage.
     if (play != NULL) {
         Enemy_StartFinishingBlow(play, &this->actor);
-        if (this->actor.draw == func_80AD8D64) {
+        if (this->actor.draw == EnKame_DrawSpikedSnapper) {
             this->actor.draw = EnKame_Draw;
         } else {
             func_800BE504(&this->actor, &this->collider);
         }
     }
+
     this->actor.bgCheckFlags &= ~BGCHECKFLAG_GROUND;
     this->actor.flags &= ~ACTOR_FLAG_TARGETABLE;
     this->actor.flags |= ACTOR_FLAG_10;
     Actor_PlaySfx(&this->actor, NA_SE_EN_PAMET_DEAD);
-    this->unk_29E = 0;
-    this->actionFunc = func_80AD825C;
+    this->timer = 0;
+    this->actionFunc = EnKame_Dead;
 }
 
-void func_80AD825C(EnKame* this, PlayState* play) {
+/**
+ * Launches the Snapper into the air and spins it around wildly. Makes the Snapper
+ * burst into flames once it touches the ground.
+ */
+void EnKame_Dead(EnKame* this, PlayState* play) {
     SkelAnime_Update(&this->snapperSkelAnime);
     if ((this->actor.bgCheckFlags & BGCHECKFLAG_GROUND) && (this->actor.velocity.y < 0.0f)) {
         Actor_PlaySfx(&this->actor, NA_SE_EN_HIPLOOP_LAND);
-        func_80AD8364(this);
+        EnKame_SetupBurstIntoFlames(this);
     } else {
-        if (this->unk_29E == 1) {
+        //! @bug unreachable code: EnKame_SetupDead sets this variable to 0, and no other code that runs
+        //! while the Snapper is in this state will change it to anything else.
+        if (this->timer == 1) {
```

Comment:
Kind of curious, did you try setting timer to something other than 0 in `SetupDead` to see what happens?

Reply:
Good question. I have tried it once, but it's a bit hard to spot because the Snapper goes flying pretty fast when it dies. So I went a little overkill and coded up a cutscene for the Snapper's death to make it more obvious. Here's the code as written:

https://github.com/zeldaret/mm/assets/12245827/da26c867-4a53-46e3-8e20-624b9349b220

And here it is with `this->timer = 1` in `EnKame_SetupDead`. In other words, this is what it would look like if you could reach the unreachable code:

https://github.com/zeldaret/mm/assets/12245827/c9efa58d-dd6f-4bd9-8bd8-d90893fd8c91

In the final game, the Snapper turns red when it gets hit, then that color filter gradually fades. Once it's fully gone, the Snapper turns "dark". If you could reach the unreachable code, the Snapper will stay red the whole time, and that red color filter won't fade.